hls4ml/backends/vivado/vivado_backend.py

```diff
@@ -376,12 +376,27 @@ def init_depconv2d(self, layer):
         )  # TODO Once we have SeparableConv implementation for io_parallel this should be set properly
         layer.set_attr('implementation', layer.model.config.get_conv_implementation(layer).lower())
 
-    def _set_pooling_accum_t(self, layer, pool_size):
+    def _set_pooling_accum_t(self, layer: Layer, pool_size):
         extra_bits = ceil_log2(pool_size)
+
+        input_layer_name = layer.inputs[0]
+        input_t = layer.model.get_layer_output_variable(input_layer_name).type
+        accum_t = layer.attributes['accum_t']
+        pool_op = layer.attributes['pool_op'].lower()
+
+        accum_t.name = f'{layer.name}_accum'
+        # accum_t.name was likely model_default, change to avoid override parameters for other by chance
+
         accum_t = layer.get_attr('accum_t')
-        accum_t.precision.width += extra_bits * 2
-        if isinstance(accum_t.precision, FixedPrecisionType):
-            accum_t.precision.integer += extra_bits
+
+        if pool_op == 'max':
```

Comment:
This should not be set here. The user can override it, in which case we should follow what the user requests. Otherwise, if this was set to "auto", then the automatic precision inference (#855) will do something similar. We can check if it does the right thing there and fix it if needed (but let me check that my updates that went in that PR first before adding code to it.)

Reply:
This is the current version that I have:  https://github.com/jmitrevs/hls4ml/blob/qonnx-1p0/hls4ml/model/optimizer/passes/infer_precision.py. I will try to extract it and make a PR to 855 with it within the next 2 hours or so.

Reply:
I made a PR to Vladimir's branch:  https://github.com/vloncar/hls4ml/pull/55 . You can probably do the same. I don't think I explicitly did anything for pooling, but there is the merge Average and Add that probably have the same logic in my PR.

Reply:
Opened PR at [#56](https://github.com/vloncar/hls4ml/pull/56). This PR will be updated if that one got merged.